pkg/rhos/securitygroups.go

```diff
@@ -0,0 +1,270 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package rhos
+
+import (
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/openstack/compute/v2/extensions/secgroups"
+	"github.com/gophercloud/gophercloud/openstack/compute/v2/servers"
+	"github.com/gophercloud/gophercloud/openstack/networking/v2/extensions/security/rules"
+	"github.com/gophercloud/gophercloud/pagination"
+	"github.com/pkg/errors"
+	"github.com/submariner-io/cloud-prepare/pkg/api"
+	"github.com/submariner-io/cloud-prepare/pkg/k8s"
+)
+
+type CloudInfo struct {
+	Client    *gophercloud.ProviderClient
+	InfraID   string
+	Region    string
+	K8sClient k8s.Interface
+}
+
+func (c *CloudInfo) openInternalPorts(infraID string, ports []api.PortSpec,
+	computeClient, networkClient *gophercloud.ServiceClient) error {
+	groupName := infraID + internalSecurityGroupSuffix
+	opts := secgroups.CreateOpts{
+		Name:        groupName,
+		Description: "Submariner Internal",
+	}
+
+	isFound, err := checkIfSecurityGroupPresent(groupName, computeClient)
+	if isFound {
+		return nil
+	}
+
+	if err != nil {
+		return err
+	}
```

Comment:
Should always check error first.

```suggestion
        if err != nil {
		return err
	}
	
	if isFound {
		return nil
	}
```